src/crepe/system/CollisionSystem.h

```diff
@@ -1,13 +1,246 @@
 #pragma once
 
+#include <vector>
+#include <variant>
+#include <optional>
+
+#include "api/Rigidbody.h"
+#include "api/Transform.h"
+#include "api/BoxCollider.h"
+#include "api/CircleCollider.h"
+#include "api/Vector2.h"
+
+#include "Collider.h"
 #include "System.h"
 
+
 namespace crepe {
 
+
+//! A system responsible for detecting and handling collisions between colliders.
 class CollisionSystem : public System {
 public:
 	using System::System;
+private:
+	
+	//! A variant type that can hold either a BoxCollider or a CircleCollider.
+	using collider_variant = std::variant<std::reference_wrapper<BoxCollider>, std::reference_wrapper<CircleCollider>>;
+
+	//! Enum representing the types of collider pairs for collision detection.
+	enum class CollisionInternalType {
+    BOX_BOX,
+    CIRCLE_CIRCLE,
+    BOX_CIRCLE,
+    CIRCLE_BOX,
+	};
+
+	/**
+		* \brief A structure to store the collision data of a single collider.
+		* 
+		* This structure stores the collider type, its associated transform, and its rigidbody.
```

Comment:
Dat kunnen we al zien :^)

Reply:
ik weet niet wat ik met deze feedback moet doen?